### PR TITLE
Restructured project file system and implemented search result display code logics 

### PR DESCRIPTION
**Pull request by yzhang54**

### CODE REVIEW

**Review by weilonp — APPROVED** [review]

**Review by dangan114 — COMMENTED** [review]
Looks good!

**Review by dangan114 — APPROVED** [review]

File: src/app/pages/browse/browse.component.ts [hunk 1]
```diff
@@ -16,18 +17,37 @@ interface Result {
...
+})
+export class UpdateRowsPipe implements PipeTransform {
+
+  transform<T>(value: T[], perRow: number): T[][] {
+    let updated_db_result: T[][] = [];
+    for (let i = 0; i < value.length; i += perRow) {
+      updated_db_result.push(value.slice(i, i + perRow))
+    }
+    return updated_db_result;
+  }
```
Comment (JOOOEG123) [inline src/app/pages/browse/browse.component.ts:32]: do you still need this after using pagination? @yzhang54

File: src/app/pages/browse/browse.component.ts [hunk 2]
```diff
@@ -209,22 +249,29 @@ export class BrowseComponent implements OnInit {
     var select_option = (<HTMLInputElement>document.getElementById("viewSelect")).value;
```
Comment (JOOOEG123) [inline src/app/pages/browse/browse.component.ts:249]: use twoway databinding

File: src/app/pages/browse/browse.component.html [hunk 3]
```diff
@@ -140,25 +142,29 @@ <h6 class="flex-grow-1 bd-highlight">Filter</h6>
...
                 <div class="m-3">| Order by: A-Z</div>
                 <div class="p-2">
-                    <select class="form-select m-0" style="max-height:40px" id="viewSelect" aria-label="View select" (change)="onViewChange()">
-                        <option selected >List</option>
-                        <option >Grid</option>
+                    <select class="form-select m-0" style="max-height:40px" id="viewSelect" aria-label="View select"
+                        (change)="onViewChange()">
+                        <option selected>List</option>
+                        <option>Grid</option>
                     </select>
```
Comment (JOOOEG123) [inline src/app/pages/browse/browse.component.html:156]: <select class="form-select m-0" style="max-height:40px" id="viewSelect" [(ngModel)]="seclectInput" aria-label="View select"
                        (change)="onViewChange()">
                        <option selected>List</option>
                        <option>Grid</option>
                    </select>

File: src/app/pages/browse/browse.component.html [hunk 4]
```diff
@@ -167,56 +173,56 @@ <h6 class="flex-grow-1 bd-highlight">Filter</h6>
         </div>
 
         <div class="d-flex  border-bottom border-danger">
-            <div class="col">
-                A
+            <div class="col text-primary">
+                {{this.currentLetter}}
```
Comment (JOOOEG123) [inline src/app/pages/browse/browse.component.html:177]: Hint (You dont have to change): **this.currentLetter** is same as **currentLetter**

File: src/app/pages/browse/browse.component.html [hunk 5]
```diff
@@ -167,56 +173,56 @@ <h6 class="flex-grow-1 bd-highlight">Filter</h6>
...
-                  <li class="page-item"><a class="page-link" href="#">Z</a></li>
-                </ul>
-              </nav>
+
+            <div class="btn-group">
+                <button class="btn btn-link" *ngFor="let letter of letters"
+                    [ngClass]="currentLetter === letter ? 'btn-danger' : ''" (click)="lettersBtnClick(letter)">
+                    {{letter}}
+                </button>
+            </div>
```
Comment (JOOOEG123) [inline src/app/pages/browse/browse.component.html:185]: Looks good.

File: src/app/pages/browse/browse.component.html [hunk 6]
```diff
@@ -167,56 +173,56 @@ <h6 class="flex-grow-1 bd-highlight">Filter</h6>
...
-                  <li class="page-item"><a class="page-link" href="#">L</a></li>
-                  <li class="page-item"><a class="page-link" href="#">M</a></li>
-                  <li class="page-item"><a class="page-link" href="#">N</a></li>
-                  <li class="page-item"><a class="page-link" href="#">O</a></li>    
-                  <li class="page-item"><a class="page-link" href="#">P</a></li>
-                  <li class="page-item"><a class="page-link" href="#">Q</a></li>
-                  <li class="page-item"><a class="page-link" href="#">R</a></li>
-                  <li class="page-item"><a class="page-link" href="#">S</a></li>
-                  <li class="page-item"><a class="page-link" href="#">T</a></li>
-                  <li class="page-item"><a class="page-link" href="#">U</a></li>
```
Comment (JOOOEG123) [inline src/app/pages/browse/browse.component.html:197]: very good change

File: src/app/pages/browse/browse.component.html [hunk 7]
```diff
@@ -167,56 +173,56 @@ <h6 class="flex-grow-1 bd-highlight">Filter</h6>
...
+                </button>
+            </div>
+
         </div>
+
         <!-- display search results -->
-        <ul *ngFor="let result of results;">
-            <div class="d-flex p-4 my-1 bg-primary rounded-start rounded-pill" name="searchLayout" (click)="searchResultClick()">
+        
+        <div *ngIf="curView === 'List'; else elseBlock">
```
Comment (JOOOEG123) [inline src/app/pages/browse/browse.component.html:191]: Good way to use else and ngTemplate.

File: src/app/pages/browse/browse.component.html [hunk 8]
```diff
@@ -253,10 +259,11 @@ <h6 class="flex-grow-1 bd-highlight">Filter</h6>
 
 
 <!-- profile -->
-<div class = "bg-dark">
+<div class="bg-dark">
 
     <div class="pt-4">
-        <div class="d-flex pb-5 pt-5 pe-5 my-5 bg-light rounded-start rounded-pill col-9 align-items-center " style="padding-left: 8em; max-height:200px">
+        <div class="d-flex pb-5 pt-5 pe-5 my-5 bg-light rounded-start rounded-pill col-9 align-items-center "
+            style="padding-left: 8em; max-height:200px">
```
Comment (JOOOEG123) [inline src/app/pages/browse/browse.component.html:266]: pb-5 && pt-5 is the same as py-5

**Review by dangan114 — APPROVED** [review]
Looks good!

**Comment by JOOOEG123** [comment]
PR looks Good. Can you please run prietter format so that the code looks clean.

**Pull request merged**